### PR TITLE
chore: specify repo settings overrides in local configuration

### PR DESCRIPTION
**Pull request by chingor13**
See https://github.com/googleapis/repo-automation-bots/pull/754

Required status check configs can now be configured locally.

### CODE REVIEW

File: .github/sync-repo-settings.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    - "linkage-monitor"
+    - "lint"
+    - "clirr"
+    - "units (7)"
+    - "units (8)"
+    - "units (11)"
+    - "Kokoro - Test: Integration"
+    - "cla/google"
+# List of explicit permissions to add (additive only)
+permissionRules: []
```
Comment (JustinBeckwith) [inline .github/sync-repo-settings.yaml:43]: I think we would want to include `yoshi-admins`, `yoshi-java`, and `yoshi-java-admins` with the appropriate permissions here.  I yanked the code that does magic broadly.

Comment (chingor13) [reply .github/sync-repo-settings.yaml:43]: Oh, ok. I saw the comment in the example and assumed the language teams were given and this was the place for the product team.

File: .github/sync-repo-settings.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# Whether or not squash-merging is enabled on this repository.
+# Defaults to `true`
+rebaseMergeAllowed: false
+
+# Whether or not rebase-merging is enabled on this repository.
+# Defaults to `true`
+squashMergeAllowed: true
```
Comment (BenWhitehead) [inline .github/sync-repo-settings.yaml:7]: The comment for these two keys appear to be flipped

Comment (chingor13) [reply .github/sync-repo-settings.yaml:7]: Yes, will fix in the example README for sync-repo-settings bot. https://github.com/googleapis/repo-automation-bots/pull/774

Comment (BenWhitehead) [reply .github/sync-repo-settings.yaml:7]: :+1:

**Pull request merged**